Database/Stored Procedures/check_performance.sql

```diff
@@ -0,0 +1,129 @@
+/*==============================================================*/
+/* DBMS name:		FestiBase									*/
+/* PDM version:		6											*/
+/* Last edited:		20-12-2017									*/
+/* Edited by:		Yuri Vannisselroy							*/
+/* Procedure:		Check performance constraints (C5)			*/
+/*==============================================================*/
+
+USE FestiBase
+GO
+
+/*
+	Procedure that checks the following:
+	- An artist cannot have 2 performance at the same time.
+	- 2 artists cannot be performing on the same stage at the same time.
+	- The performancetime must be between the festivals duration time.
+	- If the date of performance of an artist is unknown, the start time
+	  must be unknown as well.
+*/
+DROP PROC IF EXISTS sp_check_performance;
+GO
+CREATE PROC sp_check_performance
+	@performance_number		INT = NULL,
+	@artist_number			INT,
+	@podium_schedule_number	INT	= NULL,
+	@festival_number		INT,
+	@start_time				TIME = NULL,
+	@play_time				INT,
+	@min_prep_time			INT,
+	@insert					BIT
+AS
+BEGIN
+	DECLARE @break_time INT
+
+	BEGIN TRY
+		/* Is the date of performance not known? */
+		IF @podium_schedule_number IS NULL
+			BEGIN
+				/* Is the start_time known when it shouldn't be known? */
+				IF @start_time IS NOT NULL
+					BEGIN
+						;THROW 50000, '@start_time must be NULL if the podium_schedule_number is NULL', 1
+					END
+			END
+
+		/* Is the date of performance known? */
+		IF @podium_schedule_number IS NOT NULL
+			BEGIN
+				/* Is the start_time known? */
+				IF @start_time IS NOT NULL
+					BEGIN
+
+						/* Is the artists minimal preparation time longer than the break_time within that schedule? */
+						IF @min_prep_time > (SELECT break_time
+											 FROM podium_schedule
+											 WHERE podium_schedule_number = @podium_schedule_number)
```

Comment:
je zet break_time later in een variabele, waarom doe je t dan niet nu al? zo hoeft dit maar een keer uit de database gehaald te worden (regel 63). Je kan dit dan bijvoorbeeld boven de if statement doen en je kan dan IF(@min_prep_time > @break_time) doen